src/api/app/models/workflow_run.rb

```diff
@@ -27,6 +29,9 @@ class WorkflowRun < ApplicationRecord
   has_many :scm_status_reports, class_name: 'SCMStatusReport', dependent: :destroy
   has_many :event_subscriptions, dependent: :destroy
 
+  # Create the event only if the status was updated and changed from any value to 'fail'
+  after_save :create_event, if: -> { saved_change_to_attribute(:status)&.second == 'fail' }
```

Comment:
```suggestion
  after_save :create_event, if: -> { has_failed? }
  
  def has_failed?
    saved_change_to_attribute(:status)&.second == 'fail'
  end
```
I find this easier to follow. What do you think?

Reply:
Sure. We created `status_changed_to_fail?`. Please take a look again.